### PR TITLE
Add ability to put ticket num in either the PR title or the git branch name

### PR DESCRIPTION
**Pull request by neilkimmett**
We label our branches with the JIRA issue rather than the PR, so I added it as an option. I haven't managed to test this with our repo yet, but hopefully it'll work 🤞

### CODE REVIEW

**Comment by macklinu** [comment]
This is a cool idea! Thanks for contributing. Let me take a deeper look at this tonight/tomorrow and follow up. 😄

**Comment by neilkimmett** [comment]
Not sure why CI failed? Looks like it actually passed?!
image

**Comment by macklinu** [comment]
Check out these docs for the danger object API reference. Can’t remember off hand how to get the branch but hopefully this points you in the right direction https://danger.systems/js/reference.html#the-danger-dsl-root-objects

**Comment by macklinu** [comment]
also check out the `danger pr` command https://danger.systems/js/guides/the_dangerfile.html#using-danger-pr haven’t used it in a while but hopefully that helps in testing too :)

**Comment by neilkimmett** [comment]
Shockingly theres actually nothing in there to get the branch 😱. Gonna take a look at either adding it to Danger, or pulling in https://github.com/jonschlinkert/git-branch in the plugin. Thoughts?

**Comment by macklinu** [comment]
what about the value of `danger.github.pr.base`? that might be the branch that is used to open the PR

**Comment by neilkimmett** [comment]
`danger.github.head.ref` seems to do the trick! Cheers pal.

File: src/index.ts [hunk 1]
```diff
@@ -22,6 +22,11 @@ export interface Options {
    * @returns {string}
    */
   format?: (emoji: string, jiraUrls: string[]) => string;
+  /**
+   * The location of the JIRA issue, either the PR title, or the git branch
+   * Defaults to `title`
+   */
+  location?: "title" | "branch";
```
Comment (macklinu) [inline src/index.ts]: I'm sort of wondering if we should even make this an option. Part of me wants to make this plugin "smarter" or closer to "zero config" and just look in the following places for a JIRA ticket ID without requiring the user to configure this:

* `danger.github.pr.title` - PR title
* `danger.github.pr.head.ref` - the PR head branch name
* (I think that's it right now)

What do you think about that as someone who uses this plugin? Do you want more configurability, or do you want things to Just Work™?

Comment (macklinu) [reply src/index.ts]: Maybe also check all the commit messages too? Link all the things 🔗

Comment (neilkimmett) [reply src/index.ts]: Oooh that would be pretty nice actually. What would it do if it found multiple tickets; just spit out links to all of em?

Comment (neilkimmett) [reply src/index.ts]: In that vein we could remove the need to specify a JIRA key and look for anything that looks like `ABC[DEF]-123`? In the meantime how would you feel about merging this and tackling all of that as a follow up?

Comment (macklinu) [reply src/index.ts]: Yeah this can be a follow up in a future release. This is a great non-breaking addition to the plugin. 🎉

**Comment by neilkimmett** [comment]
Awesome, thanks dude. Any chance you could push a new version to NPM?

**Pull request merged**